src/arch/xtensa/CMakeLists.txt

```diff
@@ -98,8 +98,11 @@ else()
 endif()
 
 # C & ASM flags
-target_compile_options(sof_options INTERFACE ${stdlib_flag} -fno-inline-functions ${XTENSA_C_ASM_FLAGS})
-
+if(CONFIG_INLINE_FUNCTIONS)
+	target_compile_options(sof_options INTERFACE ${stdlib_flag} ${XTENSA_C_ASM_FLAGS})
```

Comment:
Minor: when -finline-functions is not passed (like we do here), then this optimization is controlled by the optimization level (-O), so this is a bit misleading. But I'm ok to go like this as this is aligned with previous practise in SOF.

Reply:
exactly, we did not add -finline_functions option,  since our current target is only for xcc, and sof will use xcc with -o3, it will enable inline_functions automatically, in future, if sof used for other platforms(like IA), this logic still works right? compiler should have this sense.